src/Inventory/Request.php

```diff
@@ -115,7 +114,6 @@ protected function handleTask($task): array
                 return $this->handleInventoryTask();
             break;
             default:
-                $this->addError("Task '$task' is not supported.", 400);
```

Comment:
Same as for unsupported queries/actions

Reply:
Actually, GLPI-Agent is trying to communicate in JSON with glpi 10-beta, so we reached this code if these tasks are enabled in agent as it just lists that tasks in its contact query.
The point here is, the server is not trying to handle a query but decide what to do to handle a task that the agent says it supports. This is not an agent error to support tasks server does still not and we should NOT answer an error on `contact` query in that case.

Reply:
So removing the header detection will fix that as well. But problem is still present: if we reach this case, it will fail silently.

Reply:
I'm not agree as this is not really to be considered as a fail, so the server still have to return a 200 on a contact if the agent announced it has enabled tasks the server doesn't support.
In my last commits, I changed anyway this behavior:
 * on unknown task, we really return a 400 error
 * on an unsupported task, we update the "message" part of the response and we set a disabled property to ask agent to not run task. And now the agent in debug mode returns this when using `--tasks=esx,collect` option with glpi-agent:
```
[info] sending contact request to server0
[debug] server message: collect task not supported;esx task not supported
```
And it won't run the related tasks.

Reply:
Anyway, this is an issue now if you have glpi-inventory plugin installed as it supports some tasks but glpi-agent won't try them is they are disabled during the contact. We should probably have a hook here so the plugin can report in the contact answer which task it supports and which protocol is supported for each task.

Reply:
> Anyway, this is an issue now if you have glpi-inventory plugin installed as it supports some tasks but glpi-agent won't try them is they are disabled during the contact. We should probably have a hook here so the plugin can report in the contact answer which task it supports and which protocol is supported for each task.

Another PR will come after that to fix this PR side-effect.